stl/inc/xstring

```diff
@@ -2432,6 +2438,33 @@ public:
         _Proxy._Release();
     }
 
+    explicit basic_string(_String_constructor_concat_tag, const basic_string& _Alsource, const _Elem* const _Left_ptr,

```

Comment:
The naming `_Alsource` is a little weird; we generally use `_Almeow` to refer to different flavors of allocators, but this is a `basic_string` serving as the source of an allocator. Any other naming would be fine, e.g. `_Source_of_alloc` or something.